src/ledger/assets/tokens.ts

```diff
@@ -65,36 +65,28 @@ export class Tokens extends Asset {
 			return "uncomparable";
 		}
 		let res = ["lt", "eq", "gt"] as const;
-		let a = this.value;
-		let b = (asset as Tokens).value;
+		let a = this.value.sort();
+		let b = (asset as Tokens).value.sort();
```

Comment:
```suggestion
		let a = this.value;
		let b = (asset as Tokens).value;
```

Reply:
And add a `requireUniquePositiveEntries` function which is used in the constructor after sorting the input array, something like:

```ts
// requireUniquePositiveEntries expects `v` to be sorted and asserts that every
// entry in `v` is unique and a positive `bigint`. If the invariant does not
// hold, an error is thrown.
function requireUniquePositiveEntries(v: bigint[]) {
	for (let k = 0; k < v.length - 1; k++) {
		if (v[k] < 0n) throw new Error("invalid token id in token set");
		if (v[k] === v[k + 1]) throw new Error("no unique token entries");
	}
}
```